src/main/java/de/tum/in/www1/artemis/web/rest/QuizSubmissionResource.java

```diff
@@ -131,6 +132,8 @@ public QuizSubmissionResource(QuizExerciseService quizExerciseService, QuizSubmi
         // the following method either reuses an existing participation or creates a new one
         StudentParticipation participation = participationService.startExercise(quizExercise, user, false);
         // we set the exercise again to prevent issues with lazy loaded quiz questions
+        participation.setInitializationState(InitializationState.FINISHED);
```

Comment:
Should this not be added after the result was saved into the database and not when submitting a practice run?

Reply:
I had a LazyLoading Error when saving it after the Exercise gets manually added to the participation a few lines above, thats why i put it before it. But saving it after the result got saved seems fine as well and makes more sense, thanks

Reply:
5f4a6cc